processor/src/test/java/org/mapstruct/ap/test/canonicalconstructor/defaultcomponentmodel/UserCanonicalConstructorMapper.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct.ap.test.canonicalconstructor.defaultcomponentmodel;
+
+import org.mapstruct.Mapper;
+import org.mapstruct.ap.test.canonicalconstructor.shared.AddressDto;
+import org.mapstruct.ap.test.canonicalconstructor.shared.AddressEntity;
+import org.mapstruct.ap.test.canonicalconstructor.shared.UserDto;
+import org.mapstruct.ap.test.canonicalconstructor.shared.UserEntity;
+import org.mapstruct.factory.Mappers;
+
+@Mapper
```

Comment:
If you add additional Mappers via `@Mapper(uses=.....)` you get errors.
`ERROR UserCanonicalConstructorMapper.java:20 No implementation was created for UserCanonicalConstructorMapper due to having a problem in the erroneous element null.`

Could you try that?

Reply:
Hey @zyberzebra, thanks for the review. Good catch, I'll look into it to fix this :sweat_smile:

Reply:
let me know if you need some help. I'm trying to fix it also but it gets kinda tricky here : `org.mapstruct.ap.internal.processor.AnnotationBasedComponentModelProcessor#process`

Reply:
i tried to implement something. It works but is kinda "dirty"
https://github.com/mapstruct/mapstruct/pull/3582